src/app/(space)/(content)/[[...pathname]]/page.tsx

```diff
@@ -35,6 +38,12 @@ export default async function Page(props: { params: PagePathParams }) {
     } = await fetchPageData(params);
     const linksContext: PageHrefContext = {};
 
+    const canFallback = !!searchParams.fallback;
+    if (!page && canFallback) {
+        const rootPage = resolveFirstDocument(pages, []);
+        rootPage?.page ? redirect(pageHref(pages, rootPage?.page, linksContext)) : notFound();
```

Comment:
Maybe we should not call the `notFound` here and let the logic below handle it.

```suggestion
        if (rootPage?.page) {
            redirect(pageHref(pages, rootPage?.page, linksContext));
        }
```